hs_collection_resource/models.py

```diff
@@ -0,0 +1,137 @@
+from lxml import etree
+
+from django.db import models
+# from django.contrib.contenttypes import generic
+from django.contrib.contenttypes.fields import GenericRelation
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor, \
+    CoreMetaData, AbstractMetaDataElement
+from hs_core.hydroshare.utils import current_site_url
+
+class CollectionResource(BaseResource):
+    objects = ResourceManager('CollectionResource')
+
+    class Meta:
+        proxy = True
+        verbose_name = 'Collection Resource'
+
+    @classmethod
+    def get_supported_upload_file_types(cls):
+        # no file types are supported
+        return ()
+
+    @classmethod
+    def can_have_multiple_files(cls):
+        # resource can't have any files
+        return False
+
+    @property
+    def metadata(self):
+        md = CollectionMetaData()
+        return self._get_metadata(md)
+
+processor_for(CollectionResource)(resource_processor)
+
+class CollectionItems(AbstractMetaDataElement):
+    term = 'CollectionItems'
+    collection_items = models.ManyToManyField(BaseResource, null=True, blank=True)
+
+    def get_collection_items_str(self):
+        return ','.join([res.title+":" + res.resource_type for res in self.collection_items.all()])
+
+    @classmethod
+    def create(cls, **kwargs):
+        if 'collection_items' in kwargs:
+            metadata_obj = kwargs['content_object']
+            new_meta_instance = CollectionItems.objects.create(content_object=metadata_obj)
+            for res_id in kwargs['collection_items']:
+                res = BaseResource.objects.filter(short_id__iexact=res_id)
+                if res.exists():
+                    new_meta_instance.collection_items.add(res[0])
+            return new_meta_instance
+        else:
+            raise ObjectDoesNotExist("No collection_items parameter was found in the **kwargs list")
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        meta_instance = CollectionItems.objects.get(id=element_id)
+        if meta_instance:
```

Comment:
This ```if``` is not needed.The previous statement will raise exception if a matching instance of CollectionItems is not found.